lib/index.js

```diff
@@ -208,14 +208,25 @@ function validateDate(date, key) {
     return date;
   }
   if(date instanceof Date) {
-    return (Math.round(date.getTime() / 1000));
-  } else if(Number.isInteger(date)) {
+    const timestamp = date.getTime() / 1000;
+    if(key === 'created') {
+      // using Math.ceil here can result
+      // in timestamps that are 1 second in the future
+      return Math.floor(timestamp);
+    }
+    if(key === 'expires') {
+      return Math.ceil(timestamp);
+    }
+    return Math.round(timestamp);
```

Comment:
Is there another key? Seems like this should be an error case.

Reply:
Addressed here: https://github.com/digitalbazaar/http-signature-header/pull/17/commits/470a57786c0b05a4a310574ef484c1e616381e05